### PR TITLE
Create new private liboqs-cupqc repository

### PR DESCRIPTION
**Pull request by praveksharma**
This creates a new private repository to work on integrating NVIDIA's cupqc in liboqs. This work shall remain private temporarily at NVIDIA's request to not expose the API calls against cupqc before it is merged into liboqs main.

Users with write access:

- @praveksharma
- @ydoroz
- @stevenireeves
- @neil-lindquist

### CODE REVIEW

**Review by ryjones — APPROVED** [review]

**Comment by ryjones** [comment]
@praveksharma please look at the change I made

**Comment by praveksharma** [comment]
Thank you @ryjones, this looks good to me!

**Review by baentsch — CHANGES_REQUESTED** [review]
@praveksharma @ryjones Do I understand this PR as having created an OQS sub project to which no OQS team member but @praveksharma has access? If so, how should we comment on this API integration? Or is no comment indeed desired until this has been merged (and we shall deal with consequences of potentially sub-optimal API decisions later on)? A bit more explanation would be welcome. Thanks.

**Comment by praveksharma** [comment]
You are correctly interpreting the changes made in this PR @baentsch . Thank you for raising your concerns.

While this private repo is made for working on these changes the eventual PR to merge the changes into liboqs would be public. Any sub-optimal choices pointed out by reviewers would then only result in a longer wait till the PR is accepted - the health of liboqs will not be affected.

Although I don't see why more OQS team members should not be involved in this integration should they wish to be. I shall speak with the NVIDIA team this week so we are on the same page about this; assuming this proposal is not faced with objection I shall solicit more participation in the next status call and on the OQS mailing list.

**Comment by praveksharma** [comment]
@ryjones I've confirmed that the changes made in this PR are accurate however @stevenireeves who is in liboqs-cupqc-maintainers is unable to access the repo (GitHub serves them the “This is not the web page you are looking for.” page). 

I'm unable to confirm access rights in repo's settings page; could you please check if stevenireeves has the correct access rights?

**Comment by stevenireeves** [comment]
> @ryjones I've confirmed that the changes made in this PR are accurate however @stevenireeves who is in liboqs-cupqc-maintainers is unable to access the repo (GitHub serves them the “This is not the web page you are looking for.” page).
> 
> I'm unable to confirm access rights in repo's settings page; could you please check if stevenireeves has the correct access rights?

Bump

**Comment by ryjones** [comment]
@stevenireeves you need to accept the pending invite

**Comment by stevenireeves** [comment]
> @stevenireeves you need to accept the pending invite

Thanks, for some reason github didn't send me an email to approve the request. I verify that I have access now...

**Comment by lakshya-chopra** [comment]
Hello,
With Nvidia's cuPQC now released, are there any plans to make this work publicly available in the near future?
I'd be happy to test it and explore how it can be integrated into our project.
Thanks :)

**Comment by baentsch** [comment]
Thanks for raising this issue. Indeed, when looking at the NVIDIA website, [it currently reads](https://developer.nvidia.com/cupqc):

> Adopted by the Global Cryptography Ecosystem
> 
> Used by leading cryptographic developers, cloud service providers, and leading security startups. Available through the open-source library liboqs.

@praveksharma @stevenireeves Could you please explain how this "availability" is provided? Should everyone be able to use the `liboqs` APIs (library?) to access cupqc? If so, how? Where is this documented? If not, why (in light of the above)?

--> Worthwhile [adding a conversation about this to the next TSC agenda](https://github.com/open-quantum-safe/tsc/blob/main/meetings/2024-12-10/agenda.md), @dstebila ? In any case, I'd propose to revert the status of this sub project immediately to "public access" to bring published statements and reality in line (and satisfy my curiosity as well as that of @lakshya-chopra :).

**Comment by stevenireeves** [comment]
@lakshya-chopra @baentsch cuPQC is now public, meaning anyone can download the library and use it. We're working on a PR to merge a liboqs implementation of ML-KEM using our library and public API soon. Infact, the development work is finished, we just need to be able to create a PR from our private repo to the public one.

**Comment by dstebila** [comment]
> we just need to be able to create a PR from our private repo to the public one.

Is there anything you need from us in order to be able to do that?

**Comment by praveksharma** [comment]
> Is there anything you need from us in order to be able to do that?

@stevenireeves and I had a meeting this past week regarding this. As discussed earlier, I'll complete a preliminary review on the private repo before a PR on the public repo is made.

**Pull request merged**
